File: binding/include/libqcborstreamwriter.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qcborstreamwriter.html
pub const qcborstreamwriter = struct {

    /// New constructs a new QCborStreamWriter object.
    pub fn New(device: ?*anyopaque) QtC.QCborStreamWriter {
        return qtc.QCborStreamWriter_new(@ptrCast(device));
    }


    /// New2 constructs a new QCborStreamWriter object.
    pub fn New2(data: []u8) QtC.QCborStreamWriter {
        const data_str = qtc.libqt_string{
    .len = data.len,
    .data = data.ptr,
};
return qtc.QCborStreamWriter_new2(data_str);
    }


    pub fn SetDevice(self: ?*anyopaque, device: ?*anyopaque) void {
        qtc.QCborStreamWriter_SetDevice(@ptrCast(self), @ptrCast(device));
    }

    pub fn Device(self: ?*anyopaque, ) QtC.QIODevice {
        return qtc.QCborStreamWriter_Device(@ptrCast(self));
    }

    pub fn Append(self: ?*anyopaque, u: u64) void {
        qtc.QCborStreamWriter_Append(@ptrCast(self), u);
    }

    pub fn Append2(self: ?*anyopaque, i: i64) void {
        qtc.QCborStreamWriter_Append2(@ptrCast(self), i);
    }

    pub fn Append3(self: ?*anyopaque, n: u64) void {
        qtc.QCborStreamWriter_Append3(@ptrCast(self), @intCast(n));
    }

    pub fn Append4(self: ?*anyopaque, ba: []u8) void {
        const ba_str = qtc.libqt_string{
    .len = ba.len,
    .data = ba.ptr,
};
qtc.QCborStreamWriter_Append4(@ptrCast(self), ba_str);
    }

    pub fn Append5(self: ?*anyopaque, str: QLatin1StringView) void {
        qtc.QCborStreamWriter_Append5(@ptrCast(self), str);
    }

    pub fn Append6(self: ?*anyopaque, str: []const u8) void {
        qtc.QCborStreamWriter_Append6(@ptrCast(self), @ptrCast(str));
    }

    pub fn Append7(self: ?*anyopaque, tag: u64) void {
        qtc.QCborStreamWriter_Append7(@ptrCast(self), @intCast(tag));
    }

    pub fn Append8(self: ?*anyopaque, tag: i32) void {
        qtc.QCborStreamWriter_Append8(@ptrCast(self), @intCast(tag));
    }

    pub fn Append9(self: ?*anyopaque, st: u8) void {
        qtc.QCborStreamWriter_Append9(@ptrCast(self), @intCast(st));
    }

    pub fn Append11(self: ?*anyopaque, f: QtC.qfloat16) void {
        qtc.QCborStreamWriter_Append11(@ptrCast(self), @ptrCast(f));
    }

    pub fn Append12(self: ?*anyopaque, f: f32) void {
        qtc.QCborStreamWriter_Append12(@ptrCast(self), f);
    }

    pub fn Append13(self: ?*anyopaque, d: f64) void {
        qtc.QCborStreamWriter_Append13(@ptrCast(self), d);
    }

    pub fn AppendByteString(self: ?*anyopaque, data: []const u8, lenVal: i64) void {
        qtc.QCborStreamWriter_AppendByteString(@ptrCast(self), @ptrCast(data), lenVal);
    }

    pub fn AppendTextString(self: ?*anyopaque, utf8: []const u8, lenVal: i64) void {
        qtc.QCborStreamWriter_AppendTextString(@ptrCast(self), @ptrCast(utf8), lenVal);
    }

    pub fn Append14(self: ?*anyopaque, b: bool) void {
        qtc.QCborStreamWriter_Append14(@ptrCast(self), b);
    }

    pub fn AppendNull(self: ?*anyopaque, ) void {
        qtc.QCborStreamWriter_AppendNull(@ptrCast(self));
    }

    pub fn AppendUndefined(self: ?*anyopaque, ) void {
        qtc.QCborStreamWriter_AppendUndefined(@ptrCast(self));
    }

    pub fn Append15(self: ?*anyopaque, i: i32) void {
        qtc.QCborStreamWriter_Append15(@ptrCast(self), i);
    }

    pub fn Append16(self: ?*anyopaque, u: u32) void {
        qtc.QCborStreamWriter_Append16(@ptrCast(self), u);
    }

    pub fn Append17(self: ?*anyopaque, str: []const u8) void {
        qtc.QCborStreamWriter_Append17(@ptrCast(self), @ptrCast(str));
    }

    pub fn StartArray(self: ?*anyopaque, ) void {
        qtc.QCborStreamWriter_StartArray(@ptrCast(self));
    }

    pub fn StartArray2(self: ?*anyopaque, count: u64) void {
        qtc.QCborStreamWriter_StartArray2(@ptrCast(self), count);
    }

    pub fn EndArray(self: ?*anyopaque, ) bool {
        return qtc.QCborStreamWriter_EndArray(@ptrCast(self));
    }

    pub fn StartMap(self: ?*anyopaque, ) void {
        qtc.QCborStreamWriter_StartMap(@ptrCast(self));
    }

    pub fn StartMap2(self: ?*anyopaque, count: u64) void {
        qtc.QCborStreamWriter_StartMap2(@ptrCast(self), count);
    }

    pub fn EndMap(self: ?*anyopaque, ) bool {
        return qtc.QCborStreamWriter_EndMap(@ptrCast(self));
    }

    pub fn Append22(self: ?*anyopaque, str: []const u8, size: i64) void {
        qtc.QCborStreamWriter_Append22(@ptrCast(self), @ptrCast(str), size);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QCborStreamWriter_Delete(@ptrCast(self));
    }
};

File: binding/src/libqcborstreamwriter.cpp
#include <QByteArray>
#include <QCborStreamWriter>
#include <QIODevice>
#include <QStringView>
#include <qcborstreamwriter.h>
#include "libqcborstreamwriter.h"
#include "libqcborstreamwriter.hxx"

QCborStreamWriter* QCborStreamWriter_new(QIODevice* device) {
	 return new QCborStreamWriter(device);
}

QCborStreamWriter* QCborStreamWriter_new2(libqt_string data) {
	 return new QCborStreamWriter(QByteArray(data.data, data.len));
}

void QCborStreamWriter_SetDevice(QCborStreamWriter* self, QIODevice* device) {
	self->setDevice(device);
}

QIODevice* QCborStreamWriter_Device(const QCborStreamWriter* self) {
	return self->device();
}

void QCborStreamWriter_Append(QCborStreamWriter* self, unsigned long long u) {
	self->append(u);
}

void QCborStreamWriter_Append2(QCborStreamWriter* self, long long i) {
	self->append(i);
}

void QCborStreamWriter_Append3(QCborStreamWriter* self, quint64 n) {
	self->append(static_cast<QCborNegativeInteger>(n));
}

void QCborStreamWriter_Append4(QCborStreamWriter* self, const libqt_string ba) {
	self->append(QByteArray(ba.data, ba.len));
}

void QCborStreamWriter_Append5(QCborStreamWriter* self, QLatin1StringView str) {
	self->append(str);
}

void QCborStreamWriter_Append6(QCborStreamWriter* self, QStringView* str) {
	self->append(*str);
}

void QCborStreamWriter_Append7(QCborStreamWriter* self, quint64 tag) {
	self->append(static_cast<QCborTag>(tag));
}

void QCborStreamWriter_Append8(QCborStreamWriter* self, int tag) {
	self->append(static_cast<QCborKnownTags>(tag));
}

void QCborStreamWriter_Append9(QCborStreamWriter* self, quint8 st) {
	self->append(static_cast<QCborSimpleType>(st));
}

void QCborStreamWriter_Append11(QCborStreamWriter* self, qfloat16* f) {
	self->append(*f);
}

void QCborStreamWriter_Append12(QCborStreamWriter* self, float f) {
	self->append(f);
}

void QCborStreamWriter_Append13(QCborStreamWriter* self, double d) {
	self->append(d);
}

void QCborStreamWriter_AppendByteString(QCborStreamWriter* self, const char* data, ptrdiff_t lenVal) {
	self->appendByteString(data, lenVal);
}

void QCborStreamWriter_AppendTextString(QCborStreamWriter* self, const char* utf8, ptrdiff_t lenVal) {
	self->appendTextString(utf8, lenVal);
}

void QCborStreamWriter_Append14(QCborStreamWriter* self, bool b) {
	self->append(b);
}

void QCborStreamWriter_AppendNull(QCborStreamWriter* self) {
	self->appendNull();
}

void QCborStreamWriter_AppendUndefined(QCborStreamWriter* self) {
	self->appendUndefined();
}

void QCborStreamWriter_Append15(QCborStreamWriter* self, int i) {
	self->append(i);
}

void QCborStreamWriter_Append16(QCborStreamWriter* self, unsigned int u) {
	self->append(u);
}

void QCborStreamWriter_Append17(QCborStreamWriter* self, const char* str) {
	self->append(str);
}

void QCborStreamWriter_StartArray(QCborStreamWriter* self) {
	self->startArray();
}

void QCborStreamWriter_StartArray2(QCborStreamWriter* self, unsigned long long count) {
	self->startArray(count);
}

bool QCborStreamWriter_EndArray(QCborStreamWriter* self) {
	return self->endArray();
}

void QCborStreamWriter_StartMap(QCborStreamWriter* self) {
	self->startMap();
}

void QCborStreamWriter_StartMap2(QCborStreamWriter* self, unsigned long long count) {
	self->startMap(count);
}

bool QCborStreamWriter_EndMap(QCborStreamWriter* self) {
	return self->endMap();
}

void QCborStreamWriter_Append22(QCborStreamWriter* self, const char* str, ptrdiff_t size) {
	self->append(str, size);
}

void QCborStreamWriter_Delete(QCborStreamWriter* self) {
    delete self;
}

